WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -163,58 +175,21 @@ final class DashboardViewModel: ObservableObject {
             group.addTask { [weak self] in
                 await self?.syncDashboardEssentialData()
             }
-            group.addTask { [weak self] in
-                guard let self else { return }
-                await reloadCards(showOnDashboardCards)
+            if dashboardCards.isNotEmpty {
+                group.addTask { [weak self] in
+                    guard let self else { return }
+                    await reloadCards(showOnDashboardCards)
+                }
             }
             group.addTask { [weak self] in
                 await self?.checkInboxEligibility()
             }
-        }
-        isReloadingAllData = false
-    }
-
-    /// Sync essential data to construct the dashboard
-    ///
-    @MainActor
-    func syncDashboardEssentialData() async {
-        await withTaskGroup(of: Void.self) { group in
             group.addTask { [weak self] in
                 guard let self else { return }
-                await self.syncAnnouncements(for: self.siteID)
-            }
-            group.addTask { [weak self] in
-                await self?.reloadBlazeCampaignView()
-            }
-            group.addTask { [weak self] in
-                await self?.updateJetpackBannerVisibilityFromAppSettings()
-            }
-            group.addTask { [weak self] in
-                await self?.updateHasOrdersStatus()
+                savedCards = await loadDashboardCards() ?? []
```

Comment:
Non-blocking suggestion: It might help to move this into a small function to have the loading and writing into `savedCards` logic inside a single method. We can call that function from here and from `onViewAppear`

Reply:
Good point! Updated the logic in 2616119ed5539e10895b5c4a12bc780b3322e2c2.